Star Wars API - Console Challenge (#111)
neuefische/web-exercises: sessions/js-fetch/star-wars-console/README.md

# Fetch Star Wars API and Log to Console

> Fetching data is awesome, yet it can be frightening at the beginning. But don't worry, we are going to make you familiar with the syntax and it will be fine ;)

Reviewer: ```suggestion
Fetching data is awesome, yet it can be frightening in the beginning. But don't worry – we are going to help you familiarize yourself with the syntax of fetch and all will be fine ;)
```

## Task

Have a look at the [index.js](js/index.js) file: There is an `url` variable and an empty `fetchData()` function.

Inside of the `fetchData()` function, use the `url` variable to fetch data from the Star Wars API.

> Log the fetched data to the console and play with them:

Reviewer: ```suggestion
Log the fetched data to the console and play around with it:
```

- log the entire data object
- log different values of the data object
- Bonus: Can you log the eye color of R2-D2?

The following hints may guide you:

- use a `try...catch` block
- make sure to wait for all asynchronous operations

> - make sure to handle bad responses from the API as well

Reviewer: ```suggestion
- make sure to handle bad responses from the API as well
- you can test bad responses via adding a random string of characters to the end of the URL stored in the url variable
```
